### PR TITLE
QA-5531 Update create table syntax, add test for AbstractSqlClient

### PR DESCRIPTION
**Pull request by Berkof**
https://ward.gridgain.com/tests/672b9822cbbc295a2bce166d
Sucessfull local run with AI3: [WARD](https://ward.gridgain.com/tests/672b9afacbbc295a2bce167a)

### CODE REVIEW

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java [hunk 1]
```diff
@@ -355,10 +334,36 @@ private void createTestTable(Ignite node0) throws DBException {
...
+  public String createTableSQL(String createZoneReq) {
+    String fieldsSpecs = fields.stream()
+        .map(e -> e + " VARCHAR")
+        .collect(Collectors.joining(", "));
+
+    String withZoneName = "";
+    if (!createZoneReq.isEmpty()) {
+      if (useColumnar) {
+        withZoneName = String.format(
+            " ZONE=\"%s\", STORAGE PROFILE '%s', SECONDARY STORAGE PROFILE '%s'",
```
Comment (nsivkov) [inline ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java:346]:
```suggestion
            " ZONE=\"%s\" STORAGE PROFILE '%s' SECONDARY STORAGE PROFILE '%s'",
```

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java [hunk 2]
```diff
@@ -57,47 +57,46 @@
...
-  protected static final String HOSTS_PROPERTY = "hosts";
+  protected int fieldCount;
 
-  protected static final String PRIMARY_COLUMN_NAME = "ycsb_key";
-
-  protected static final String DEFAULT_ZONE_NAME = "Z1";
+  protected String fieldPrefix;
 
-  protected static final String DEFAULT_STORAGE_PROFILE_NAME = "default";
+  protected long recordsCount;
```
Comment (iartiukhov) [inline ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java:80]: I do not think that it is a good idea to remove `static` access modifier from declaration of fields. It is used as a simple way to share the same field across all benchmarking threads. Most YCSB benchmarks for other DBs follow the same pattern. So let's keep that.

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java [hunk 3]
```diff
@@ -355,10 +334,36 @@ private void createTestTable(Ignite node0) throws DBException {
...
+            DEFAULT_ZONE_NAME,
+            storageProfile,
+            secondaryStorageProfile);
+      } else {
+        withZoneName = String.format(" ZONE \"%s\";", DEFAULT_ZONE_NAME);
+      }
+    }
+
+    String createTableReq = String.format("CREATE TABLE IF NOT EXISTS %s(%s  VARCHAR PRIMARY KEY, %s) %s", cacheName,
+        PRIMARY_COLUMN_NAME, fieldsSpecs, withZoneName);
```
Comment (nsivkov) [inline ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java:356]:
```suggestion
    String createTableReq = String.format("CREATE TABLE IF NOT EXISTS %s(%s VARCHAR PRIMARY KEY, %s)%s", cacheName,
        PRIMARY_COLUMN_NAME, fieldsSpecs, withZoneName);
```

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java [hunk 4]
```diff
@@ -355,10 +334,36 @@ private void createTestTable(Ignite node0) throws DBException {
...
+    String withZoneName = "";
+    if (!createZoneReq.isEmpty()) {
+      if (useColumnar) {
+        withZoneName = String.format(
+            " ZONE=\"%s\", STORAGE PROFILE '%s', SECONDARY STORAGE PROFILE '%s'",
+            DEFAULT_ZONE_NAME,
+            storageProfile,
+            secondaryStorageProfile);
+      } else {
+        withZoneName = String.format(" ZONE \"%s\";", DEFAULT_ZONE_NAME);
```
Comment (nsivkov) [inline ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java:351]:
```suggestion
        withZoneName = String.format(" ZONE \"%s\"", DEFAULT_ZONE_NAME);
```

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java [hunk 5]
```diff
@@ -370,19 +375,16 @@ private String createZoneSQL() {
...
     String paramPartitions = partitions.isEmpty() ? "" : "partitions=" + partitions;
-    String params = Stream.of(paramStorageProfiles, paramReplicas, paramPartitions)
+
+    String params = Stream.of(paramStorageProfile, paramSecondaryStorageProfile, paramReplicas, paramPartitions)
         .filter(s -> !s.isEmpty())
         .collect(Collectors.joining(", "));
-    String reqWithParams = params.isEmpty() ? "" : " WITH " + params;
 
-    String createZoneReq = "CREATE ZONE IF NOT EXISTS " + DEFAULT_ZONE_NAME + reqWithParams + ";";
+    String createZoneReq = String.format("CREATE ZONE IF NOT EXISTS %s WITH %s;", DEFAULT_ZONE_NAME, params);
```
Comment (nsivkov) [inline ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java:387]: There should be logic for such case:
```
CREATE ZONE IF NOT EXISTS Z1 WITH STORAGE_PROFILES='default,myColumnarStore'
```

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java [hunk 6]
```diff
@@ -370,19 +375,16 @@ private String createZoneSQL() {
...
-        storageProfile;
-
-    String paramStorageProfiles = String.format("STORAGE_PROFILES='%s'", storageProfiles);
+    String paramStorageProfile = String.format("STORAGE PROFILE '%s'", storageProfile);
+    String paramSecondaryStorageProfile = String.format("SECONDARY STORAGE PROFILE '%s'", secondaryStorageProfile);
     String paramReplicas = replicas.isEmpty() ? "" : "replicas=" + replicas;
     String paramPartitions = partitions.isEmpty() ? "" : "partitions=" + partitions;
-    String params = Stream.of(paramStorageProfiles, paramReplicas, paramPartitions)
+
+    String params = Stream.of(paramStorageProfile, paramSecondaryStorageProfile, paramReplicas, paramPartitions)
```
Comment (iartiukhov) [inline ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java:383]: We should not create a zone with `SECONDARY STORAGE PROFILE` if `useColumnar` is `false.

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java (hunk 2 above)
Comment (iartiukhov) [reply ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java:80]: Also, the `recordsCount` variable is initialized only once in the `initProperties` method. So only 1 thread among N will have this field initialized.

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java (hunk 6 above)
Comment (Berkof) [reply ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java]: My bad, zone creation reverted

File: ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java (hunk 2 above)
Comment (Berkof) [reply ignite3/src/main/java/site/ycsb/db/ignite3/IgniteAbstractClient.java:80]: But we need it as instance variables to make unit test. I've move initProperties before the init lock to initialize that properties on each instance creation.

**Pull request merged**